lib/Optimizer/CodeGen/WireSetsToProfileQIR.cpp

```diff
@@ -371,6 +388,29 @@ struct WireSetToProfileQIRPrepPass
     LLVM_DEBUG(llvm::dbgs() << "Module after prep:\n"; op->dump());
   }
 };
+
+struct WireSetToProfileQIRPostPass
+    : public cudaq::opt::impl::WireSetToProfileQIRPostBase<
+          WireSetToProfileQIRPostPass> {
+  using WireSetToProfileQIRPostBase::WireSetToProfileQIRPostBase;
+
+  void runOnOperation() override {
+    ModuleOp op = getOperation();
+    auto *ctx = &getContext();
+    RewritePatternSet patterns(ctx);
+    QuakeTypeConverter quakeTypeConverter;
+    patterns.insert<WireSetRewrite>(quakeTypeConverter, ctx);
+    ConversionTarget target(*ctx);
+    target.addLegalDialect<arith::ArithDialect, cudaq::cc::CCDialect,
+                           func::FuncDialect, LLVM::LLVMDialect>();
```

Comment:
Not sure you really need this since it is removing Ops and never adds anything new.

Reply:
Removed in https://github.com/NVIDIA/cuda-quantum/pull/2069/commits/ab08dfe310358446a6addc6290057060e64f4145